opensearch-project/OpenSearch-Dashboards
# Fix data source info is missing in default query when click Discover from other pages

### Description

When we click Discover sidebar from other pages, we don't fetch the data source info and enrich the default dataset, so that some unexpected behavior happens.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8582

## Screenshot

No UI change.

The direct impact is that when click `Discover` from other pages, the selected default dataset always shows the data source prefix, and makes other components work.

<img width="1234" alt="image" src="https://github.com/user-attachments/assets/f58b81e4-1be3-4ca1-a97d-4e8f437775f4">


## Testing the changes
1. Create an index pattern in an external data source, and set it as the default dataset.
2. Click `Discover` from other pages, check if the selected dataset shows the data source prefix, and also check if the url contains data source info.


## Changelog



### Check List
